docs/developer/python-openflow.rst

```diff
@@ -0,0 +1,206 @@
+
+.. code:: python
+
+  >>> import pyof
+  >>> pyof.v0x01.symmetric.hello.Hello()
+
+  AttributeError: module 'pyof' has no attribute 'v0x01'
+
+
+.. code:: python
+
+  >>> old_dir = dir() + ['old_dir']
+
+  >>> from pyof import *
+  >>> from pyof.v0x01 import *
+  >>> from pyof.v0x01.symmetric import *
+
+  >>> [i for i in dir() if i not in old_dir]
+  []
+
+nothing is imported...
+
+pyof Objects
+++++++++++++
+pyof objects are instances of GenericStruct or GenericType or other classes
+derived from these.
+They provide the methods ``pack`` ``unpack`` and ``get_size``.
+
+initialization
+++++++++++++++
+``GenericType``s accept as argument an initial value.
+``GenericStruct``s derivatives usually accept as argument initial values
+for its class attributes.
+
+packing
++++++++
+To pack a pyof object, simply call its ``pack()`` method, which returns a
+``bytes`` object of the binary representation.
+
+unpacking
++++++++++
+To unpack a buffer as pyof object, simply initialize a new object and
+call its ``unpack()`` method, passing the buffer as the argument. The
+buffer will be unpacked in place setting the attributes of the object
+instance used to make the call.
+
+
+New definitions
+===============
+``python-openflow`` provides some base classes and basic type classes meant to
+be used in a new message/struct class definition.
+
+Difference between "structs" and "messages"
++++++++++++++++++++++++++++++++++++++++++++
+Messages are GenericStruct derivatives who carry a header attribute containing
+an OpenFlow header struct.
+
+Unlike ``GenericStruct``s their unpack method accepts a buffer argument with
+its packed body (without the associated header).
+
+How to code a new struct/message
+++++++++++++++++++++++++++++++++
+To implement a new struct, you need to:
+- define a new class which derives from ``GenericStruct``;
+- define class attributes in pack order (with optional initial values) whose
+  type are pyof objects themselves.
+
+Struct example:
+
+.. code:: python
+
+    >>> from pyof.foundation.base import GenericStruct
+    >>> from pyof.foundation.basic_types import UBInt8, UBInt16
+
+    >>> class MyNewStruct(GenericStruct):
+            my_first_attribute = UBInt8(255)
+            my_second_attribute = UBInt16(0)
+            my_third_attribute = UBInt8(255)
+
+which can be used like this:
+
+.. code:: python
+
+    >>> my_struct = MyNewStruct()
+    >>> my_struct.my_second_attribute = 2
+    >>> p = my_struct.pack()
+    >>> print(p)
+    b'\xff\x00\x02\xff'
+
+    >>> my_struct_2 = MyNewStruct()
+    >>> my_struct_2.unpack(p)
+    >>> my_struct_2.my_second_attribute
+    UBInt16(2)
+
+Message example:
+
+.. code:: python
+
+    >>> from pyof.foundation.base import GenericMessage
+    >>> from pyof.foundation.basic_types import UBInt8, UBInt16
+    >>> from pyof.v0x04.common.header import Header
+
+    >>> class MyNewMessage(GenericMessage):
+            header = Header(message_type=255)
+            my_struct_field = MyNewStruct()
```

Comment:
It is defined in the previous example, and it seems very  clear to me. I was avoiding being redundant.